anta/tests/snmp.py

```diff
@@ -237,3 +240,121 @@ def test(self) -> None:
+              notification_type: trap
+              version: v1
+              udp_port: 162
+              community_string: public
+              user: public
+    ```
+    """
+
+    name = "VerifySNMPNotificationHost"
+    description = "Verifies the SNMP notification host (SNMP manager) configurations."
+    categories: ClassVar[list[str]] = ["snmp"]
+    commands: ClassVar[list[AntaCommand | AntaTemplate]] = [AntaCommand(command="show snmp notification host", revision=1)]
+
+    class Input(AntaTest.Input):
+        """Input model for the VerifySNMPNotificationHost test."""
+
+        notification_hosts: list[SNMPHost]
+        """List of SNMP hosts."""
+
+        class SNMPHost(BaseModel):
+            """Model for a SNMP Manager."""
+
+            hostname: IPv4Address
+            """IP address of the SNMP notification host."""
+            vrf: str = "default"
+            """Optional VRF for SNMP Hosts. If not provided, it defaults to `default`."""
+            notification_type: Literal["trap", "inform"]
+            """Type of SNMP notification (trap or inform)."""
+            version: Literal["v1", "v2c", "v3"]
+            """SNMP protocol version."""
+            udp_port: Port | int = 162
+            """UDP port for SNMP. If not provided then defaults to 162."""
+            community_string: str | None = None
+            """Optional SNMP community string for authentication."""
+            user: str | None = None
+            """Optional SNMP user for authentication."""
+
+            @model_validator(mode="after")
+            def validate_inputs(self: BaseModel) -> BaseModel:
+                """Validate the inputs provided to the SNMPHost class.
+
+                If SNMP version is either v1 or v2c, community string must be provided.
+
+                If SNMP version is v3, user must be provided.
+                """
+                if self.version in ["v1", "v2c"] and self.community_string is None:
+                    msg = "Community string must be provided when SNMP Protocol version is either v1 or v2c."
+                    raise ValueError(msg)
+                if self.version == "v3" and self.user is None:
+                    msg = "User must be provided when SNMP Protocol version is v3."
+                    raise ValueError(msg)
+                return self
+
+    @AntaTest.anta_test
+    def test(self) -> None:
+        """Main test function for VerifySNMPNotificationHost."""
+        failures: str = ""
+
+        for host in self.inputs.notification_hosts:
+            hostname = str(host.hostname)
+            vrf = host.vrf
+            version = host.version
+            notification_type = host.notification_type
+            udp_port = host.udp_port
+            community_string = host.community_string
+            user = host.user
+
+            # Verify SNMP host details.
+            if not (host_details := get_item(self.instance_commands[0].json_output["hosts"], "hostname", hostname)):
```

Comment:
we can get empty list in case no SNMP host configured so in that case just break with a general message

Reply:
Added check. Thanks!!